aws_sagemaker_studio/sagemaker_studio_image_build/tensorflow_bring_your_own/tensorflow_bring_your_own.ipynb

```diff
@@ -0,0 +1,740 @@
+    "                \"iam:GetRole\",\n",
+    "                \"iam:ListRoles\"\n",
+    "            ],\n",
+    "            \"Resource\": \"*\"\n",
+    "        },\n",
+    "        {\n",
+    "            \"Effect\": \"Allow\",\n",
+    "            \"Action\": \"iam:PassRole\",\n",
+    "            \"Resource\": \"arn:aws:iam::*:role/*\",\n",
+    "            \"Condition\": {\n",
+    "                \"StringLikeIfExists\": {\n",
+    "                    \"iam:PassedToService\": \"codebuild.amazonaws.com\"\n",
+    "                }\n",
+    "            }\n",
+    "        }\n",
+    "    ]\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# SageMaker Execution Role \n",
+    "import sagemaker\n",
+    "import boto3 \n",
+    "\n",
+    "try:\n",
+    "    role = sagemaker.get_execution_role()\n",
+    "except:\n",
+    "    role = get_execution_role()\n",
+    "\n",
+    "print(\"Using IAM role arn: {}\".format(role))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Step 3: Building and registering the container using the SageMaker Studio Image Build CLI\n",
+    "\n",
+    "We will now create our training container image, using the SageMaker Studio Image Build CLI  \n",
+    "\n",
+    "To do this we need to navigate to the directory containing our Dockerfile and simply execute the build command: \n",
+    "         \n",
+    "                  sm-docker build .\n",
+    "                  \n",
+    "The build command can optionally take additional arguments depending on your needs:\n",
+    "\n",
+    "                  sm-docker build . --file /path/to/Dockerfile --build-arg foo=bar\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!cd container && sm-docker build . "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    " **NOTE** Make sure you copy the Image URI output above as we will need to identify that training image as input into our training job\n",
+    " \n",
+    "---"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Download the CIFAR-10 dataset\n",
+    "Our training algorithm is expecting our training data to be in the file format of [TFRecords](https://www.tensorflow.org/guide/datasets), which is a simple record-oriented binary format that many TensorFlow applications use for training data.\n",
+    "Below is a Python script adapted from the [official TensorFlow CIFAR-10 example](https://github.com/tensorflow/models/tree/master/tutorials/image/cifar10_estimator), which downloads the CIFAR-10 dataset and converts them into TFRecords."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install ipywidgets"
```

Comment:
This is out of place... and might be a problem w/ JupyterLab/Studio.
I was tracking issues with this module somewhere else.
https://stackoverflow.com/questions/49542417/how-to-get-ipywidgets-working-in-jupyter-lab
Should fix the install to use sys, but also move it to the top where the other pip stuff happens... or if an optional sequence, put it where it gets used.

Reply:
There is a dependency within the generate_cifar records script in /utils [Based on original notebook in advanced functionality].  Added clarity around the dependency and validated functionality for next push.